[#129] UserService 코틀린으로 전환

hanwix2 opened the pull request:
#129 

## 작업내용
- `UserService.java` -> `UserService.kt`
- 코틀린 관련 의존성 추가
- UserService와 관련된 모델(DTO) 코틀린으로 변환
    - 대상: `UserRequest`, `UserResponse`, `UserSession`, `UserUpdate`
    - data class 적용
    - 코틀린 적용에 따른 빌더패턴 미사용
- 리팩토링에 따른 관련 테스트 코드 변경

## ETC
-  IntelliJ IDE 변환기로 코틀린 코드로 변환 후 최적화
- 기존 자바 코드에서 롬복(lombok)을 사용해 생성한 메소드를 코틀린 코드에선 호출할 수 없었음 
    > DTO 클래스를 코틀린으로 변경
-  IDE 환경에서는 자바에서 코틀린으로 변환된 코드가 기존 위치에 있어도 빌드 성공하지만, IDE 없이는 빌드 실패 문제 발생
    - 공식 문서에 소개된 방법으로 각각의 코드를 별도로 분리하여 컴파일 시점을 달리하도록 적용
    > https://kotlinlang.org/docs/maven.html#compile-kotlin-and-java-sources

hanwix2 committed:
[#129] UserService java to kotlin 변환

- 인텔리제이 자체 변환기 사용

hanwix2 committed:
[#129] UserService 코틀린 코드 수정 및 최적화

hanwix2 committed:
[#129] UserSession 코틀린으로 변경

- UserSession.java -> UserSession.kt

hanwix2 committed:
[#129] UserSession properties에 기본 접근제어자 적용

- getter 사용을 위함

hanwix2 committed:
[#129] UserUpdateRequest 코틀린으로 변환

hanwix2 committed:
[#129] UserRequest - 코틀린으로 변경

hanwix2 committed:
[#129] UserResponse - 코틀린으로 변환

- UserRequest getter 사용을 위한 기본 접근 제어자로 properties 선언

hanwix2 commented:
자바 코드를 코틀린으로의 변경에 따라 관련 DTO가 변경되고, 기존 **Lombok**을 사용하던 코드의 충돌이 발생하여 처리중입니다.

f-lab-dev commented on src/main/java/com/show/showticketingservice/service/UserService.kt:
```diff
@@ -0,0 +1,72 @@
+    }
+
+    @Transactional(readOnly = true)
+    fun checkIdExists(userId: String?) {
```
`userId`가 nullable일 필요가 있을까요?

f-lab-dev commented on src/main/java/com/show/showticketingservice/service/UserService.kt:
```diff
@@ -0,0 +1,72 @@
+        }
+    }
+
+    fun getUser(userIdRequest: String?, passwordRequest: String): UserResponse {
```
변수명에서 Request는 빼는게 나을 것 같습니다~ 변수의 역할을 파악하는데에 약간의 방해가 될 것 같네요

hanwix2 committed:
[#129] UserService 관련 모델, 테스트 코드 최적화

- 모델(dto): UserRequest, UserResponse, UserSession, UserUpdateRequest
    - data class로 변경
    - 코틀린에서 Validation을 사용하기 위해 target(field) 지정
- 테스트 코드
    - 코틀린으로 변환된 dto는 builder 패턴을 사용하지 않음에 따라 관련
      코드 수정(새로운 객체 할당)
    - dto 생성시 null을 허용하지 않으므로 관련 코드 삭제

hanwix2 committed:
[#129] UserService 메소드 매개변수의 null 허용 제거

- 불필요한 nullable

hanwix2 committed:
[#129] UserService getUser 메소드의 매개변수명 변경

- 변수의 역할을 명확히 알 수 있도록 필요한 정보만을 명시

hanwix2 committed:
[#129] 자바와 코틀린 코드를 별도 디렉토리로 분리

- IDE 환경에서는 자바에서 코틀린으로 변환된 코드가 기존 위치에 있어도
  빌드 성공
- 하지만 IDE 없이는 빌드 실패
- 공식 문서에 소개된 방법으로 각각의 코드를 별도로 분리하여 컴파일
  시점을 달리하도록 적용

hanwix2 committed:
[#129] Update pom.xml

Kotlin 코드를 먼저 컴파일하고 Java 코드에서 참조하기 위해 maven-compiler-plugin 관련 설정 변경
